Tool/FixtureGenerator.php

```diff
@@ -325,9 +325,29 @@ public function generateFixtureItemStub($item)
         $ids = $this->getRelatedIdsForReference($class, $item);
 
         $code = "";
+        $constructorParams = [];
         $reflexion = new \ReflectionClass($item);
         $properties = $this->getRecursiveProperties($reflexion);
-        $newInstance = $reflexion->newInstance();
+        if (null !== $reflexion->getConstructor()) {
+            foreach ($reflexion->getConstructor()->getParameters() as $parameter) {
+                if ($parameter->isDefaultValueAvailable()) {
+                    continue;
+                }
+                $paramName = $parameter->getName();
+                if ($reflexion->hasMethod('get'.ucfirst($paramName))) {
+                    $constructorParams[] = $item->{'get'.ucfirst($paramName)}();
+                } elseif ($reflexion->hasProperty($paramName)) {
+                    $reflectionProperty = $reflexion->getProperty($paramName);
+                    $reflectionProperty->setAccessible(true);
+                    $constructorParams[] = $reflectionProperty->getValue($item);
+                }
+            }
+        }
+        if (!empty($constructorParams)) {
```

Comment:
The change increase the method complexity.
please move code filling $constructorParams into its own method.
$constructorParams = $this->getConstructorParams(...);